evm/bridge-token-factory/contracts/BridgeTokenFactory.sol

```diff
@@ -242,10 +245,10 @@ contract BridgeTokenFactory is
     ) internal virtual {}
 
     function claimNativeFee(bytes calldata signatureData, BridgeTypes.ClaimFeePayload memory payload) external {
-        bytes memory borshEncodedNonces = Borsh.encodeUint32(uint32(payload.nonces.length));
+        bytes memory borshEncodedNonces = abi.encodePacked(Borsh.encodeUint32(uint32(payload.nonces.length)));
 
         for (uint i = 0; i < payload.nonces.length; ++i) {
-            uint128 nonce = payload.nonces[i];
+            uint64 nonce = payload.nonces[i];
```

Comment:
And I already forgot, is it destinationNonce or  originNonce? Anyway, should we also check the chain_id?

Reply:
It doesn't matter, will be removed